### PR TITLE
[Snyk] Fix for 11 vulnerabilities

### PR DESCRIPTION
**Pull request by techcobweb**
<p>This PR was automatically created by Snyk using the credentials of a real user.</p><br /><h3>Snyk has created this PR to fix one or more vulnerable packages in the `maven` dependencies of this project.</h3>

#### Changes included in this PR

- Changes to the following files to upgrade the vulnerable dependencies to a fixed version:
    - galasa-simplatform-application/galasa-simplatform-webapp/pom.xml
    - galasa-simplatform-application/pom.xml

##### With an upgrade:
Severity                   | Priority Score (*)                   | Issue                   | Upgrade                   | Breaking Change                   | Exploit Maturity
:-------------------------:|-------------------------|:-------------------------|:-------------------------|:-------------------------|:-------------------------
medium severity  |  **539/1000**  <br/> **Why?** Has a fix available, CVSS 6.5  | Deserialization of Untrusted Data <br/>[SNYK-JAVA-COMGOOGLECODEGSON-1730327](https://snyk.io/vuln/SNYK-JAVA-COMGOOGLECODEGSON-1730327) |  `com.google.code.gson:gson:` <br> `2.8.6 -> 2.8.9` <br>  `dev.galasa:dev.galasa.zos3270.manager:` <br> `0.15.0 -> 0.25.0` <br>  |  No  | No Known Exploit 
medium severity  |  **651/1000**  <br/> **Why?** Mature exploit, Has a fix available, CVSS 5.3  | Directory Traversal <br/>[SNYK-JAVA-COMMONSIO-1277109](https://snyk.io/vuln/SNYK-JAVA-COMMONSIO-1277109) |  `dev.galasa:dev.galasa.zos3270.manager:` <br> `0.15.0 -> 0.25.0` <br>  |  No  | Mature 
low severity  |  **399/1000**  <br/> **Why?** Has a fix available, CVSS 3.7  | Man-in-the-Middle (MitM) <br/>[SNYK-JAVA-LOG4J-1300176](https://snyk.io/vuln/SNYK-JAVA-LOG4J-1300176) |  `dev.galasa:dev.galasa.zos3270.manager:` <br> `0.15.0 -> 0.25.0` <br>  |  No  | No Known Exploit 
medium severity  |  **651/1000**  <br/> **Why?** Proof of Concept exploit, Has a fix available, CVSS 6.6  | Arbitrary Code Execution <br/>[SNYK-JAVA-LOG4J-2316893](https://snyk.io/vuln/SNYK-JAVA-LOG4J-2316893) |  `dev.galasa:dev.galasa.zos3270.manager:` <br> `0.15.0 -> 0.25.0` <br>  |  No  | Proof of Concept 
high severity  |  **619/1000**  <br/> **Why?** Has a fix available, CVSS 8.1  | SQL Injection <br/>[SNYK-JAVA-LOG4J-2342645](https://snyk.io/vuln/SNYK-JAVA-LOG4J-2342645) |  `dev.galasa:dev.galasa.zos3270.manager:` <br> `0.15.0 -> 0.25.0` <br>  |  No  | No Known Exploit 
high severity  |  **619/1000**  <br/> **Why?** Has a fix available, CVSS 8.1  | Deserialization of Untrusted Data <br/>[SNYK-JAVA-LOG4J-2342646](https://snyk.io/vuln/SNYK-JAVA-LOG4J-2342646) |  `dev.galasa:dev.galasa.zos3270.manager:` <br> `0.15.0 -> 0.25.0` <br>  |  No  | No Known Exploit 
high severity  |  **619/1000**  <br/> **Why?** Has a fix available, CVSS 8.1  | Deserialization of Untrusted Data <br/>[SNYK-JAVA-LOG4J-2342647](https://snyk.io/vuln/SNYK-JAVA-LOG4J-2342647) |  `dev.galasa:dev.galasa.zos3270.manager:` <br> `0.15.0 -> 0.25.0` <br>  |  No  | No Known Exploit 
medium severity  |  **509/1000**  <br/> **Why?** Has a fix available, CVSS 5.9  | Denial of Service (DoS) <br/>[SNYK-JAVA-LOG4J-3358774](https://snyk.io/vuln/SNYK-JAVA-LOG4J-3358774) |  `dev.galasa:dev.galasa.zos3270.manager:` <br> `0.15.0 -> 0.25.0` <br>  |  No  | No Known Exploit 
critical severity  |  **811/1000**  <br/> **Why?** Proof of Concept exploit, Has a fix available, CVSS 9.8  | Deserialization of Untrusted Data <br/>[SNYK-JAVA-LOG4J-572732](https://snyk.io/vuln/SNYK-JAVA-LOG4J-572732) |  `dev.galasa:dev.galasa.zos3270.manager:` <br> `0.15.0 -> 0.25.0` <br>  |  No  | Proof of Concept 
high severity  |  **579/1000**  <br/> **Why?** Has a fix available, CVSS 7.3  | Out-of-bounds Write <br/>[SNYK-JAVA-ORGAPACHEBCEL-3106013](https://snyk.io/vuln/SNYK-JAVA-ORGAPACHEBCEL-3106013) |  `dev.galasa:dev.galasa.zos3270.manager:` <br> `0.15.0 -> 0.25.0` <br>  |  No  | No Known Exploit 
medium severity  |  **479/1000**  <br/> **Why?** Has a fix available, CVSS 5.3  | Improper Input Validation <br/>[SNYK-JAVA-ORGAPACHEHTTPCOMPONENTS-1048058](https://snyk.io/vuln/SNYK-JAVA-ORGAPACHEHTTPCOMPONENTS-1048058) |  `org.apache.httpcomponents:fluent-hc:` <br> `4.5.12 -> 4.5.13` <br>  |  No  | No Known Exploit 

(*) Note that the real score may have changed since the PR was raised.

Check the changes in this PR to ensure they won't cause issues with your project.

------------

**Note:** *You are seeing this because you or someone else with access to this repository has authorized Snyk to open fix PRs.*

For more information:  
🧐 [View latest project report](https://app.snyk.io/org/techcobweb/project/68510268-1268-4835-a443-654375c0e331?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr)

🛠 [Adjust project settings](https://app.snyk.io/org/techcobweb/project/68510268-1268-4835-a443-654375c0e331?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr/settings)

📚 [Read more about Snyk's upgrade and patch logic](https://support.snyk.io/hc/en-us/articles/360003891078-Snyk-patches-to-fix-vulnerabilities)

[//]: # (snyk:metadata:{"prId":"d39897ee-3717-441e-8b06-38d7e13f9e68","prPublicId":"d39897ee-3717-441e-8b06-38d7e13f9e68","dependencies":[{"name":"com.google.code.gson:gson","from":"2.8.6","to":"2.8.9"},{"name":"dev.galasa:dev.galasa.zos3270.manager","from":"0.15.0","to":"0.25.0"},{"name":"org.apache.httpcomponents:fluent-hc","from":"4.5.12","to":"4.5.13"}],"packageManager":"maven","projectPublicId":"68510268-1268-4835-a443-654375c0e331","projectUrl":"https://app.snyk.io/org/techcobweb/project/68510268-1268-4835-a443-654375c0e331?utm_source=github&utm_medium=referral&page=fix-pr","type":"user-initiated","patch":[],"vulns":["SNYK-JAVA-COMGOOGLECODEGSON-1730327","SNYK-JAVA-COMMONSIO-1277109","SNYK-JAVA-LOG4J-1300176","SNYK-JAVA-LOG4J-2316893","SNYK-JAVA-LOG4J-2342645","SNYK-JAVA-LOG4J-2342646","SNYK-JAVA-LOG4J-2342647","SNYK-JAVA-LOG4J-3358774","SNYK-JAVA-LOG4J-572732","SNYK-JAVA-ORGAPACHEBCEL-3106013","SNYK-JAVA-ORGAPACHEHTTPCOMPONENTS-1048058"],"upgrade":["SNYK-JAVA-COMGOOGLECODEGSON-1730327","SNYK-JAVA-COMMONSIO-1277109","SNYK-JAVA-LOG4J-1300176","SNYK-JAVA-LOG4J-2316893","SNYK-JAVA-LOG4J-2342645","SNYK-JAVA-LOG4J-2342646","SNYK-JAVA-LOG4J-2342647","SNYK-JAVA-LOG4J-3358774","SNYK-JAVA-LOG4J-572732","SNYK-JAVA-ORGAPACHEBCEL-3106013","SNYK-JAVA-ORGAPACHEHTTPCOMPONENTS-1048058"],"isBreakingChange":false,"env":"prod","prType":"fix","templateVariants":["priorityScore"],"priorityScoreList":[539,651,399,651,619,619,619,509,811,579,479],"remediationStrategy":"vuln"})

---

**Learn how to fix vulnerabilities with free interactive lessons:**

 🦉 [Deserialization of Untrusted Data](https://learn.snyk.io/lesson/insecure-deserialization/?loc&#x3D;fix-pr)
 🦉 [Directory Traversal](https://learn.snyk.io/lesson/improper-input-validation/?loc&#x3D;fix-pr)
 🦉 [Arbitrary Code Execution](https://learn.snyk.io/lesson/malicious-code-injection/?loc&#x3D;fix-pr)
 🦉 [More lessons are available in Snyk Learn](https://learn.snyk.io/?loc&#x3D;fix-pr)

### CODE REVIEW

File: galasa-simplatform-application/pom.xml [hunk 1]
```diff
@@ -88,7 +88,7 @@
 		<dependency>
 			<groupId>dev.galasa</groupId>
 			<artifactId>dev.galasa.zos3270.manager</artifactId>
-			<version>0.15.0</version>
+			<version>0.25.0</version>
```
Comment (techcobweb) [inline galasa-simplatform-application/pom.xml:91]: should this be set to 0.28.0 (the latest version)?

Comment (techcobweb) [reply galasa-simplatform-application/pom.xml:91]: In fact, shouldn't the BOM be used instead of an explicit version ?

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Pull request closed without merging**